spec/wasCrawlDissemination/cdx_merge_sort_publish_service_spec.rb

```diff
@@ -75,23 +81,35 @@
       @cdx_backup_dir = "#{@stacks_path}/data/indices/cdx_backup"
     end
     let(:druid) { 'ii111ii1111'}
-    it 'should clean all the working directory and move the files to backup' do
+    it 'moves cdx files to cdx backup directory (from  source_dir)' do
       FileUtils.cp_r("#{cdx_file_path}/ii/.", "#{@cdx_working_dir}")
       mergeSortPublishService = Dor::WASCrawl::CDXMergeSortPublishService.new(druid, @cdx_working_dir, @cdx_backup_dir)
-      mergeSortPublishService.clean
 
+      expect(File.exist?(mergeSortPublishService.instance_variable_get(:@source_cdx_dir))).to eq true
+      expect(File.exist?("#{@cdx_working_dir}/#{druid}_merged_index.cdx")).to eq true
+      expect(File.exist?("#{@cdx_working_dir}/#{druid}_sorted_duplicate_index.cdx")).to eq true
+      expect(File.exist?("#{@cdx_backup_dir}/#{druid}")).to eq false
+      expect(File.exist?("#{@cdx_backup_dir}/#{druid}/file1.cdx")).to eq false
+      expect(File.exist?("#{@cdx_backup_dir}/#{druid}/file2.cdx")).to eq false
+
+      mergeSortPublishService.clean
+      expect(File.exist?(mergeSortPublishService.instance_variable_get(:@source_cdx_dir))).to eq false
+      expect(File.exist?("#{@cdx_working_dir}/#{druid}_merged_index.cdx")).to eq false
+      expect(File.exist?("#{@cdx_working_dir}/#{druid}_sorted_duplicate_index.cdx")).to eq false
       expect(File.exist?("#{@cdx_backup_dir}/#{druid}")).to eq true
       expect(File.exist?("#{@cdx_backup_dir}/#{druid}/file1.cdx")).to eq true
       expect(File.exist?("#{@cdx_backup_dir}/#{druid}/file2.cdx")).to eq true
+      # merged_index.cdx and sorted_duplicate_index.cdx are not kept
       expect(File.exist?("#{@cdx_backup_dir}/#{druid}_merged_index.cdx")).to eq false
       expect(File.exist?("#{@cdx_backup_dir}/#{druid}_sorted_duplicate_index.cdx")).to eq false
     end
 
-    it 'should move files to backup when the druid directory already exists' do
+    it 'moves cdx files to cdx backup directory when the druid directory already exists' do
       FileUtils.cp_r("#{cdx_file_path}/ii/.", "#{@cdx_backup_dir}")
-      expect(File.exist?("#{@cdx_backup_dir}/#{druid}")).to eq true
+
       mergeSortPublishService = Dor::WASCrawl::CDXMergeSortPublishService.new(druid, @cdx_working_dir, @cdx_backup_dir)
-      mergeSortPublishService.clean
```

Comment:
got explanation from @ndushay, i had missed the invocation in the `expect` below.